tools/src/icon4py/tools/py2fgen/wrappers/diffusion_wrapper.py

```diff
@@ -282,6 +306,15 @@ def diffusion_run(
         rho=rho,
     )
 
+    backend = diffusion_wrapper_state["granule"]._backend
+    if hdef_ic is None:
+        hdef_ic = gtx.zeros(w.domain, dtype=w.dtype, allocator=backend)
+    if div_ic is None:
+        div_ic = gtx.zeros(w.domain, dtype=w.dtype, allocator=backend)
+    if dwdx is None:
+        dwdx = gtx.zeros(w.domain, dtype=w.dtype, allocator=backend)
+    if dwdy is None:
+        dwdy = gtx.zeros(w.domain, dtype=w.dtype, allocator=backend)
```

Comment:
They are constructed in every diffusion step. We should cache them... They are only needed because GT4Py doesn't support optionals.